typhon/objects/networking/endpoints.py

```diff
@@ -268,23 +268,27 @@ def connectionStreamCB(uv_server, status):
 
 
 @autohelp
-class TCP4ServerEndpoint(Object):
+class TCPServerEndpoint(Object):
     """
-    A TCPv4 server endpoint.
+    TCP Server Endpoint.
     """
 
-    def __init__(self, port):
+    def __init__(self, port, inet_type):
         self.port = port
+        self.inet_type = inet_type
 
     def toString(self):
-        return u"<endpoint (IPv4, TCP): %d>" % (self.port,)
+        return u"<endpoint (IPv%s, TCP): %d>" % (self.inet_type, self.port)
 
     @method("Any", "Any")
     def listen(self, handler):
         vat = currentVat.get()
         uv_server = ruv.alloc_tcp(vat.uv_loop)
         try:
-            ruv.tcpBind(uv_server, "0.0.0.0", self.port)
+            if self.inet_type == 4:
```

Comment:
It would actually be even better if we fixed this API right here and now, by factoring this capability. OTOH the "fix" rapidly becomes a pile of things that should just straight-up replace this API. I've opened #146 to track this.